test/functional/rpc_net.py

```diff
@@ -139,8 +145,25 @@ def _test_getaddednodeinfo(self):
         # check that a non-existent node returns an error
         assert_raises_rpc_error(-24, "Node has not been added", self.nodes[0].getaddednodeinfo, '1.1.1.1')
 
-    def _test_getpeerinfo(self):
+    def test_getpeerinfo(self):
+        self.log.info("Test getpeerinfo")
+        # Send a transaction, generate and sync to test last_transaction values.
+        self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+        self.nodes[1].generate(1)
+        self.sync_all()
+        time_now = int(time.time())
         peer_info = [x.getpeerinfo() for x in self.nodes]
+        # Verify last_block and last_transaction values.
+        for n in range(self.num_nodes):
+            assert_approx(peer_info[n][0]['last_block'], time_now, vspan=60)
+            assert_equal(peer_info[n][1]['last_block'], 0)
+            assert_equal(peer_info[0][n]['last_transaction'], 0)
+        if peer_info[1][0]['last_transaction'] == 0:
+            assert_equal(peer_info[1][0]['last_transaction'], 0)
+            assert_approx(peer_info[1][1]['last_transaction'], time_now, vspan=60)
+        else:
+            assert_equal(peer_info[1][1]['last_transaction'], 0)
+            assert_approx(peer_info[1][0]['last_transaction'], time_now, vspan=60)
```

Comment:
Note, the `assert_approx` assertions pass locally for me with vspan values < 5 (seconds). Made them 60 to leave a wide margin for avoiding data races.